### ISSUE TITLE
NaN loss

### ISSUE THREAD

**Issue by BRO-HAMMER** [ktjonsson/keras-ArcFace#2]
Hello @ktjonsson, thank you for the implementation.
I was trying this out with different backbones, and I'm getting a very fast loss decrease and accuracy increase in cifar10, until loss becomes NaN.

When I stop training before loss is Nan, the separation between embeddings is not good (2 clusters for cifar10, for example).

Is it possible that the loss implementation may be flawed in some way? I believe that labels are leaking somehow.

Thank you!

Edit: loss is expecting y_true as a one_hot_encoded vector right?

**Comment by BRO-HAMMER** [comment]
OK, just in case somebody find this helpful:
 
I somehow fixed the problem with the NaNs by lowering the learning rate. If the learning rate is too high, log(0) can appear in the loss calculation, maybe this could be prevented by adding an epsilon to the logits or doing some clipping.

I also found a bug in the accuracy calculation:
line 76 in arcface_layer.py -> accuracy = categorical_accuracy(y_true=labels, y_pred=logits) 
should be 
**accuracy = categorical_accuracy(y_true=y_true, y_pred=logits)**
categorical_accuracy expects two one-hot encoded vectors and "labels" is a sparse vector (class indices).

Edit:
Finally solved the NaNs issue with the following change in line 52 of arcface_layer.py 
sin_t2 = tf.subtract(1., cos_t2, name='sin_2')
changed to
**sin_t2 = K.clip(tf.subtract(1.0, cos_t2, name='sin_2'), K.epsilon(), 1.0)**
